docs/instructlab-cli-1.0.0.md

```diff
@@ -0,0 +1,35 @@
+# The Road to 1.0.0
+
+_Or: How I Learned to Stop Worrying and Love to GA_
+
+## Context and Goals
+
+At the time of this writing, the InstructLab CLI has gone from a scrappy research project to an upstream community serving as the basis for multiple downstreams, with the goal to continuing to evolve the community to encourage more participation from additional stakeholders. To wit, it would behoove us to determine what exactly we should be roadmapping between now and a proper 1.0.0 release, which demonstrates the following to existing and potential community members:
```

Comment:
One thing that is missing from this context section that came up in the discussion of the SDG refactoring draft dev doc is the fact that the repo used to be `instructlab/cli` and largely focused on just the CLI layer and now it is `instructlab/instructlab` and is more of a core integrating element that contains a lot of the glue functionality for all of InstructLab.  I think this document should either state this or link to a document that states this.

I also have a proposal that I think might be more controversial.  Can we stop calling this component the CLI and provide a new name for it?   The repo could still be `instructlab/instructlab` but we would need to update the README, the upstream Slack channel name, start using the new name in dev docs and Jira issues, update the labels in meeting invites, etc.; so it is not a small undertaking, but I think it is one that will provide a lot of value in the long term because having accurate names for the major components makes it easier for new participants in the project to understand the system.

I think InstructLab core would be a good name, but I would be open to other names that convey a sense that the is the central orchestrator for all of the components of InstructLab .

Finally, I really think the name issue should be sorted out in this dev doc because settling on a name and providing consistent communication around the name should be a prereq for declaring the repo 1.0.0.

Reply:
> One thing that is missing from this context section that came up in the discussion of the SDG refactoring draft dev doc is the fact that the repo used to be instructlab/cli and largely focused on just the CLI layer and now it is instructlab/instructlab and is more of a core integrating element that contains a lot of the glue functionality for all of InstructLab. I think this document should either state this or link to a document that states this.

I agree with this sentiment, though candidly I don't know where the decision of record is for this - I'll need to do some digging.

> I also have a proposal that I think might be more controversial. Can we stop calling this component the CLI and provide a new name for it? The repo could still be instructlab/instructlab but we would need to update the README, the upstream Slack channel name, start using the new name in dev docs and Jira issues, update the labels in meeting invites, etc.; so it is not a small undertaking, but I think it is one that will provide a lot of value in the long term because having accurate names for the major components makes it easier for new participants in the project to understand the system.

I can add a section about this and update the existing references - I agree that the "CLI" is really just a part of the overall `instructlab` package that serves as a means for controlling the orchestration, which will in the future have other entrypoints such as a GUI and direct API calls. That being said, I don't feel an entirely holistic renaming of things here is necessary - some things like parts of the README and the meeting name yes, but the CLI portion of this component is always going to exist and need a place for discussion/way to be referred to.

> I think InstructLab core would be a good name, but I would be open to other names that convey a sense that the is the central orchestrator for all of the components of InstructLab .

Engine might also be a good term for it

> Finally, I really think the name issue should be sorted out in this dev doc because settling on a name and providing consistent communication around the name should be a prereq for declaring the repo 1.0.0.

Let's definitely flesh this out in the document - I'll add a new section for it. That being said, I am wary this issue might bog down the rest of the document, and reserve the right to postpone a final decision to after merging this, while specifying that a decision _must_ be made prior to 1.0.0.